### PR TITLE
Refactor Avro code for Kafka templates

### PR DESCRIPTION
**Pull request by AnandInguva**

### CODE REVIEW

File: v2/kafka-common/src/main/java/com/google/cloud/teleport/v2/kafka/transforms/KafkaRecordToGenericRecordFailsafeElementFn.java [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+    // Unified setup logic
+    if (schemaRegistryConnectionUrl != null && !schemaRegistryConnectionUrl.isBlank()) {
+      this.schemaRegistryClient =
+          new CachedSchemaRegistryClient(schemaRegistryConnectionUrl, DEFAULT_CACHE_CAPACITY);
+      this.kafkaDeserializer = new KafkaAvroDeserializer(this.schemaRegistryClient);
+    } else if (schema != null && messageFormat.equals("AVRO_BINARY_ENCODING")) {
+      this.binaryDeserializer = new BinaryAvroDeserializer(schema);
+    } else if (schema != null && messageFormat.equals("AVRO_CONFLUENT_WIRE_FORMAT")) {
+      this.schemaRegistryClient = new MockSchemaRegistryClient();
+      this.schemaRegistryClient.register(topicName, schema, 1, 1);
```
[the remark this answers is not in the document]
Comment (AnandInguva) [reply v2/kafka-common/src/main/java/com/google/cloud/teleport/v2/kafka/transforms/KafkaRecordToGenericRecordFailsafeElementFn.java:73]: the current tests passed for Kafka to BQ. Would that cover this case?

[the remark this answers is not in the document]
Comment (Dippatel98) [reply v2/kafka-common/src/main/java/com/google/cloud/teleport/v2/kafka/transforms/KafkaRecordToGenericRecordFailsafeElementFn.java:73]: Yeah, that works

File: v2/kafka-common/src/main/java/com/google/cloud/teleport/v2/kafka/transforms/AvroTransform.java [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+/**
+ * The {@link AvroTransform} class is a {@link PTransform} which transforms incoming Kafka Message
+ * objects into {@link GenericRecord} using an Avro schema file (.avsc).
+ */
+public class AvroTransform
+    extends PTransform<
+        PCollection<KafkaRecord<byte[], byte[]>>,
+        PCollection<FailsafeElement<KafkaRecord<byte[], byte[]>, GenericRecord>>> {
+
+  private String topicName;
```
[the remark this answers is not in the document]
Comment (AnandInguva) [reply v2/kafka-common/src/main/java/com/google/cloud/teleport/v2/kafka/transforms/AvroTransform.java]: Not used. removed it.

File: v2/kafka-to-gcs/src/test/java/com/google/cloud/teleport/v2/templates/TestClass.java
Comment (Dippatel98) [inline v2/kafka-to-gcs/src/test/java/com/google/cloud/teleport/v2/templates/TestClass.java:1]: What is this class used for?

Comment (AnandInguva) [reply v2/kafka-to-gcs/src/test/java/com/google/cloud/teleport/v2/templates/TestClass.java:1]: oh this needs to be removed. I used this for testing.

**Review by Dippatel98 — APPROVED** [review]

**Pull request merged**